nsxt/resource_nsxt_policy_edge_cluster.go

```diff
@@ -0,0 +1,406 @@
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/nsxt/metadata"
+)
+
+var policyEdgeClusterSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"site_path": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			Description:  "Path to the site this Host Transport Node belongs to",
+			Optional:     true,
+			ForceNew:     true,
+			Default:      defaultInfraSitePath,
+			ValidateFunc: validatePolicyPath(),
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"enforcement_point": {
+		Schema: schema.Schema{
+			Type:        schema.TypeString,
+			Description: "ID of the enforcement point this Host Transport Node belongs to",
+			Optional:    true,
+			ForceNew:    true,
+			Default:     "default",
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"inter_site_forwarding_enabled": {
+		Schema: schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+		Metadata: metadata.Metadata{
+			SdkFieldName: "InterSiteForwardingEnabled",
+		},
+	},
+	"edge_cluster_profile": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validatePolicyPath(),
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "EdgeClusterProfile",
+		},
+	},
+	"allocation_rule": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"action_based_on_failure_domain_enabled": {
+						Schema: schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType: "bool",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			Skip: true,
+		},
+	},
+	"policy_edge_node": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"edge_transport_node_path": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Required:     true,
```

Comment:
I see that `Edge transport node path can't be changed once member is added to cluster.` But I assume those can be added/removed? In which case `ForceNew` might not be applicable?

Reply:
IDK which implications `ForceNew` will imply - e.g, if I have a tier1 which is allocated on any node within the cluster, and `ForceNew` will attempt to remove the whole cluster, I'm guessing that this transaction will fail, right?

There is some API for Edge node replacement, `/policy/api/v1/infra/sites/{site-id}/enforcement-points/{enforcementpoint-id}/edge-clusters/{edge-cluster-id}/action/replace-edge-transport-node`

But IDK how we could leverage this in the provider...

Reply:
Probably best to leave as is